src/lib/visualization/pqp_visualizer.cpp

```diff
@@ -139,15 +139,15 @@ void PQPVisualizer::_build_dataflow(const std::shared_ptr<const AbstractOperator
                                     const std::shared_ptr<const AbstractOperator>& to, const InputSide side) {
   VizEdgeInfo info = _default_edge;
 
-  const auto& performance_data = from->performance_data();
-  if (performance_data.executed && performance_data.has_output) {
+  const auto& performance_data = from->performance_data;
+  if (performance_data->executed && performance_data->has_output) {
     std::stringstream stream;
-    stream << std::to_string(performance_data.output_row_count) + " row(s)/";
-    stream << std::to_string(performance_data.output_chunk_count) + " chunk(s)";
+    stream << std::to_string(performance_data->output_row_count) + " row(s)/";
+    stream << std::to_string(performance_data->output_chunk_count) + " chunk(s)";
     info.label = stream.str();
   }
 
-  info.pen_width = performance_data.output_row_count;
+  info.pen_width = performance_data->output_row_count;
```

Comment:
Didn't look at the PR yet, just something that came up when I looked at SF=10 graphs and was counting the number of digits in `59986052 row(s)`. Can we add thousands separators? Should be as easy as to throw `std::numpunct` on it.

Reply:
That’s already a TODO in the following PQP PR. 
Numpunct ain’t that straightforward though because afaik it requires a locale (I think the default C locale might be usable, but I need to take a deeper look into it).